sdk/rust/nym-sdk/src/mixnet/client.rs

```diff
@@ -45,8 +52,26 @@ impl MixnetClientBuilder {
     }
 
     #[must_use]
-    pub fn config(mut self, config: Config) -> Self {
-        self.config = Some(config);
+    pub fn gateway(mut self, user_chosen_gateway: String) -> Self {
```

Comment:
Looking good! Probably want some docs above this method just to clarify the difference between this and `fn gateway_config()` below

Now that I'm thinking about it, maybe it should be called `fn request_gateway()`? Or something similar?

And/or, `fn gateway_config()` should be renamed to `fn registered_gateway()`?

Reply:
I've applied your proposed changes and also added some more (simple) documentation on the other builder methods.